pyface/ui_traits.py

```diff
@@ -186,6 +188,54 @@ def info(self):
         )
 
 
+# -------------------------------------------------------------------------------
+#  Font
+# -------------------------------------------------------------------------------
+
+
+class PyfaceFont(TraitType):
+    """ A Trait which casts strings to a Pyface Font value.
+    """
+
+    #: The default value should be a tuple (factory, args, kwargs)
+    default_value_type = DefaultValue.callable_and_args
+
+    #: The parser to use when converting text to keyword args.  This should
+    #: accept a string and return a dictionary of Font class trait values (ie.
+    #: "family", "size", "weight", etc.).
+    parser = None
+
+    def __init__(self, value=None, *, parser=simple_parser, **metadata):
+        if parser is not None:
+            self.parser = parser
+        if value is not None:
+            font = self.validate(None, None, value)
```

Comment:
It looks as though this means that a validation error would be turned into a `TraitError`. Would it make sense to explicitly raise `ValueError` or `TypeError` (as appropriate) instead? The error message right now isn't as helpful as it could be:

```python
>>> PyfaceFont(b"times new roman")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/mdickinson/Enthought/ETS/pyface/pyface/ui_traits.py", line 212, in __init__
    font = self.validate(None, None, value)
  File "/Users/mdickinson/Enthought/ETS/pyface/pyface/ui_traits.py", line 231, in validate
    self.error(object, name, value)
  File "/Users/mdickinson/.venvs/pyface/lib/python3.9/site-packages/traits/base_trait_handler.py", line 74, in error
    raise TraitError(
traits.trait_errors.TraitError: None
```

Reply:
There might be something that's worth looking into in Traits about why the error message isn't good, since calling `validate` with `None` object and `None` trait name is done in a few places.  It feels like this code might need an extra branch for that situation: https://github.com/enthought/traits/blob/d22ce1f096e2a6f87c78d7f1bb5bf0abab1a18ff/traits/trait_errors.py#L66-L94

I'll convert this to a `ValueError` here for now.